packages/web-frontend/cypress/e2e/dashboardReports.spec.js

```diff
@@ -3,12 +3,31 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import reportUrls from '../config/dashboardReports.json';
+import { urls as reportUrls } from '../config/dashboardReports.json';
 import { SNAPSHOTS } from '../constants';
 import { preserveUserSession } from '../support';
 
-const checkResponseHasData = response =>
-  response?.body?.value !== undefined || response?.body?.data?.length > 0;
+const checkHasMatrixData = body => {
+  const { rows = [], columns = [] } = body;
+
+  const hasColumnCategories = columns[0]?.columns;
+  const getColumnKeys = cols => cols.map(c => c.key);
+  const columnKeys = hasColumnCategories
+    ? columns.map(c => getColumnKeys(c.columns)).flat()
+    : getColumnKeys(columns);
+
+  return rows.some(row => Object.keys(row).some(key => columnKeys.includes(key)));
```

Comment:
This matrix one feels a little fragile (i.e. if we changed the data structure this would also need to change), but fortunately we will find out as soon as it breaks by virtue of it being part of the e2e test suite!

Reply:
Indeed and indeed!